<?xml version="1.0"?>

<launch>

    <node name="joy_node" pkg="joy" type="joy_node" output="screen" clear_params="true">
        <param name="dev" type="str" value="/dev/input/js1"/>
    </node>
    

        <!-- ////////////////////////////// PLATFORM DETECTION NODE ////////////////////////////////////-->
    <!--
    <node name="platform_detection_real" pkg="vision" type="platform_detection_real" output="screen">
        <param name="cam_info_topic" value="/ardrone/bottom/camera_info"/>
        <param name="platform_position_topic" value="/platform/estimated_position"/>
        <param name="ardrone_navdata" value="/ardrone/navdata"/>
        <param name="imu_topic" value="/ardrone/imu"/>
    </node>
    -->


    <!-- /////////////////////////////////// ARUCO NODE ///////////////////////////////////////////-->
    <arg name="markerId"        default="7"/>
    <arg name="markerSize"      default="0.256"/>    <!-- in m -->
    <!-- <arg name="eye"             default="left"/> -->
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="ref_frame"       default="/odom"/>  <!-- leave empty and the pose will be published wrt param parent_name -->

    <node pkg="aruco_ros" type="single" name="aruco_single">
        <remap from="/camera_info" to="/ardrone/bottom/camera_info" />
        <remap from="/image" to="/ardrone/bottom/image_raw" />
        <param name="image_is_rectified" value="False"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="/ardrone_base_bottomcam"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
    </node>



    <param name="H_min" value="153"/>
    <param name="H_max" value="179"/>
    <param name="S_min" value="152"/>
    <param name="S_max" value="255"/>
    <param name="V_min" value="205"/>
    <param name="V_max" value="255"/>


    <node name="platform_detection_real" pkg="vision" type="platform_detection_real" output="screen">
        <!--TARGET SIZE-->
        <param name="target_size" value="0.25"/> <!--Value in m2-->
        <param name="altura_platform" value="0.8"/> <!--Value in m-->

        <param name="min_num_sides" value="3"/>
        <param name="max_num_sides" value="4"/>

        <param name="iterations" value="3"/>
         
        <!--TOPICS-->
        <param name="platform_position_topic" value="/platform/estimated_position"/>
        <param name="image_topic" value="/ardrone/bottom/image_raw"/>
        <param name="cam_info_topic" value="/ardrone/bottom/camera_info"/>
        <param name="ardrone_navdata" value="/ardrone/navdata"/>        
        <param name="imu_topic" value="/ardrone/raw_imu"/>
    </node>


    <param name="searching_altitude" value="1.2" /> <!-- in m -->
    <param name="takeoff_sleep_duration" value="3" /> <!-- in s -->
    <param name="takeoff_vel_z" value="0.8" /> <!-- in m/s -->


    <node name="platform_tracking_real" pkg="vision" type="platform_tracking_real" output="screen">
        <param name="Kp" value="0.04"/>
    	<param name="Ki" value="0.0"/>
    	<param name="Kd" value="0.01"/>

        <!--TOPICS-->
        <param name="cmd_topic_vel" value="/cmd_vel"/>
        <param name="platform_position_topic" value="/aruco_single/pose"/>
        <param name="ardrone_navdata" value="/ardrone/navdata"/>
        
   		<param name="takeoff_topic" value="/ardrone/takeoff" />
        <param name="landing_topic" value="/ardrone/land" />
        <param name="emergency_topic" value="/ardrone/emergency_topic" />
        <param name="tracking_topic" value="/ardrone/tracking_topic" />
    </node>





 </launch>
